src/test/scala/uk/gov/nationalarchives/tdr/api/service/FileServiceSpec.scala

```diff
@@ -96,6 +109,38 @@ class FileServiceSpec extends AnyFlatSpec with MockitoSugar with Matchers with S
     captor.getAllValues.get(0).head.userid should equal(userId)
   }
 
+  "createFile" should "create the correct static metadata" in {
+    val fixedUuidSource = new FixedUUIDSource()
+    val uuid = UUID.randomUUID()
+    val fileId = UUID.randomUUID()
+    val consignmentId = UUID.randomUUID()
+
+    val fileRepositoryMock = mock[FileRepository]
+    val consignmentRepositoryMock = mock[ConsignmentRepository]
+    val fileMetadataRepositoryMock = mock[FileMetadataRepository]
+
+    val mockFileResponse = Future.successful(List(FileRow(fileId, consignmentId, uuid, Timestamp.from(Instant.now))))
+    when(fileRepositoryMock.addFiles(any[List[FileRow]])).thenReturn(mockFileResponse)
+    val mockConsignmentResponse = Future.successful(())
+    when(consignmentRepositoryMock.addParentFolder(consignmentId, "Parent folder name")).thenReturn(mockConsignmentResponse)
+    val mockFileMetadataResponse = Future.successful(Seq(FilemetadataRow(UUID.randomUUID(), fileId, "value", Timestamp.from(Instant.now), uuid, "name")))
+
+    val captor: ArgumentCaptor[Seq[FilemetadataRow]] = ArgumentCaptor.forClass(classOf[Seq[FilemetadataRow]])
+    when(fileMetadataRepositoryMock.addFileMetadata(captor.capture())).thenReturn(mockFileMetadataResponse)
+
+    val fileService = new FileService(fileRepositoryMock, consignmentRepositoryMock, fileMetadataRepositoryMock, FixedTimeSource, fixedUuidSource)
+    fileService.addFile(AddFilesInput(consignmentId, 1, "Parent folder name"), uuid).futureValue
+
+    val metadataRows: Seq[Tables.FilemetadataRow] = captor.getValue
+    metadataRows.size should equal(staticMetadataProperties.size)
+    staticMetadataProperties.foreach(property => {
```

Comment:
Could we check each value individually rather than iterating over the list?

It feels like the test is repeating the logic of the code, and it would be nice to have a clear test stating what the expected default values are.

Reply:
Yeah good point, I've changed it to explicitly check for each of the names and values.